haystack/evaluation/eval.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any, Dict, List, Union
+
+import orjson
+
+from haystack import Pipeline, component
+from haystack.evaluation.eval_utils import (
+    convert_component_outputs_from_dict,
+    convert_component_outputs_to_dict,
+    convert_pipeline_outputs_from_dict,
+    convert_pipeline_outputs_to_dict,
+)
+
+
+class EvaluationResult:
+    """
+    EvaluationResult keeps track of all the information related to evaluation, namely the runnable (Pipeline or component), inputs, outputs, and expected outputs.
+    The EvaluationResult keeps track of all the information stored by eval.
+
+    :param runnable: The runnable (Pipeline or component) used for evaluation.
+    :param inputs: List of inputs used for evaluation.
+    :param outputs: List of outputs generated by the runnable.
+    :param expected_outputs: List of expected outputs used for evaluation.
+    """
+
+    def __init__(
+        self,
+        runnable: Union[Pipeline, component],
```

Comment:
The type is actually wrong, `component` is the decorator. What you want is `Component`, the protocol that all Component classes must respect.

You can import it with `from haystack.core.component import Component`.